### PR TITLE
Agharta/eip145 bitwise shift

### PR DESCRIPTION
**Pull request by dziabko**
Implemented EIP 145 (SAR, SHR, SHL opcodes). Implemented instruction_test.go file for basic tests. Added IsAghartha check.

### CODE REVIEW

File: core/vm/gas.go [hunk 1]
```diff
@@ -123,61 +123,64 @@ type req struct {
...
+	GAS:          {0, GasQuickStep, 1},
+	BLOCKHASH:    {1, GasExtStep, 1},
+	BALANCE:      {1, new(big.Int), 1},
+	EXTCODESIZE:  {1, new(big.Int), 1},
+	EXTCODECOPY:  {4, new(big.Int), 0},
+	SLOAD:        {1, big.NewInt(50), 1},
+	SSTORE:       {2, new(big.Int), 0},
+	SHA3:         {2, big.NewInt(30), 1},
+	CREATE:       {3, big.NewInt(32000), 1},
 	RETURNDATASIZE: {0, GasQuickStep, 1},
```
Comment (austinabell) [inline core/vm/gas.go]: I know this is a small nitpick but can you move this line to below the GAS opcode to keep the GasQuickStep costs together and format the code

File: core/vm/opcodes.go [hunk 2]
```diff
@@ -517,6 +523,9 @@ var stringToOp = map[string]OpCode{
 	"STATICCALL":     STATICCALL,
 	"REVERT":         REVERT,
 	"SUICIDE":        SUICIDE,
+	"SHL":          SHL,
```
Comment (austinabell) [inline core/vm/opcodes.go]: format please

**Review by austinabell — CHANGES_REQUESTED** [review]
Looks great! just some minor formatting and should be good

**Review by austinabell — APPROVED** [review]
utack

**Review by GregTheGreek — APPROVED** [review]

**Review by amerameen — APPROVED** [review]

**Pull request merged**